src/components/TaskDetail.jsx

```diff
@@ -0,0 +1,299 @@
+                  placeholder="Enter label text"
+                  className="label-input"
+                  maxLength={50}
+                  autoFocus
+                />
+                <div className="color-options">
+                  {LABEL_COLORS.map(color => (
+                    <button
+                      key={color}
+                      className={`color-option ${color === selectedColor ? 'selected' : ''}`}
+                      style={{ backgroundColor: color }}
+                      onClick={() => setSelectedColor(color)}
+                    />
+                  ))}
+                </div>
+                <button 
+                  className="save-label-button"
+                  onClick={handleAddLabel}
+                  disabled={!newLabelText.trim()}
+                >
+                  Add Label
+                </button>
+              </div>
+            )}
+            
+            <div className="labels-list">
+              {(task.labels || []).map(label => (
+                <div 
+                  key={label.id} 
+                  className="label-item"
+                  style={{ backgroundColor: label.color }}
+                >
+                  <span>{label.text}</span>
+                  {user && (
+                    <button
+                      className="remove-label-button"
+                      onClick={() => onRemoveLabel(task.id, label.id)}
+                      title="Remove label"
+                    >
+                      ×
+                    </button>
+                  )}
+                </div>
+              ))}
+              {(!task.labels || task.labels.length === 0) && (
+                <div className="no-labels-message">
+                  No labels added yet
+                </div>
+              )}
+            </div>
+          </div>
+
+          <div className="task-description">
+            <div className="description-header">
+              <h3>Description</h3>
+              {user && !showDeleteConfirm && (
+                <button 
+                  className="edit-button"
+                  onClick={() => setIsEditing(!isEditing)}
+                >
+                  {isEditing ? 'Cancel' : 'Edit'}
+                </button>
+              )}
+            </div>
+
+            {user && isEditing && !showDeleteConfirm ? (
+              <div className="description-editor" data-color-mode={isDark ? "dark" : "light"}>
+                <MDEditor
+                  value={description}
+                  onChange={setDescription}
+                  preview="live"
+                  height={200}
+                  hideToolbar={false}
+                  enableScroll={true}
+                  style={{
+                    backgroundColor: isDark ? '#27272A' : '#ffffff',
+                  }}
+                  previewOptions={{
+                    rehypePlugins: [[rehypeSanitize]],
+                  }}
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
previewOptions={{
  rehypePlugins: [[rehypeSanitize, {
    attributes: {
      ...rehypeSanitize.defaults.attributes,
      code: ['className'],
      span: ['className']
    },
    protocols: {
      href: ['http', 'https', 'mailto']
    }
  }]],
}}
```
The MDEditor component's rehypeSanitize configuration is basic. Consider implementing a more comprehensive sanitization configuration to prevent XSS attacks.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;